include/task_wdt/task_wdt.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Task watchdog header file
+ *
+ * This header file declares prototypes for the task watchdog APIs.
+ *
+ * The task watchdog can be used to monitor correct operation of individual
+ * threads. It can be used together with a hardware watchdog as a fallback.
+ */
+
+#ifndef TASK_WDT_H_
+#define TASK_WDT_H_
+
+#include <zephyr/types.h>
+#include <kernel.h>
+#include <device.h>
+
+/**
+ * @brief Task Watchdog APIs
+ * @defgroup task_wdt_api Task Watchdog APIs
+ * @ingroup subsystem
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Task watchdog callback. */
+typedef void (*task_wdt_callback_t)(int channel_id, void *user_data);
+
+/**
+ * @brief Initialize task watchdog.
+ *
+ * This function sets up necessary kernel timers and the hardware watchdog (if
+ * desired as fallback). It has to be called before task_wdt_add() and
+ * task_wdt_feed().
+ *
+ * @param hw_wdt Pointer to the hardware watchdog device used as fallback.
+ *               Pass NULL if no hardware watchdog fallback is desired.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP If assigning a hardware watchdog is not supported.
+ */
+int task_wdt_init(const struct device *hw_wdt);
```

Comment:
Should this take a hardware watchdog channel number as well?

Reply:
Currently, `wdt_install_timeout` (for the hardware watchdog) is called inside this function, so that's also where the channel number is assigned. The application should not do anything with the hardware watchdog (except for assigning the one that it wants to use).